test/unit/models/edition_taxonomy_tag_form_test.rb

```diff
@@ -18,17 +20,45 @@ class EditionTaxonomyTagFormTest < ActiveSupport::TestCase
     form = EditionTaxonomyTagForm.load(content_id)
 
     assert_equal(form.edition_content_id, content_id)
-    assert_equal(form.taxons, taxons)
+    assert_equal(form.selected_taxons, taxons)
     assert_equal(form.previous_version, 1)
   end
 
-  test '#publish should send a patch links to publishing-api' do
+  test '#publish should send a patch links to publishing-api with the most specific selected taxons' do
```

Comment:
This test reads a little mysteriously, because you have to know the taxonomy from the `EducationTaxonomyHelper`. Would it be better to extract the find-childest algorithm ([taxons_to_publish](https://github.com/alphagov/whitehall/pull/2997/files#diff-57ff9fc7321e84a22556b3a38f7c0017R28)) into a small class that we can test independently?